test/Microsoft.Extensions.WebSockets.Internal.Tests/WebSocketConnectionExtensions.cs

```diff
@@ -8,11 +8,14 @@ namespace Microsoft.Extensions.WebSockets.Internal.Tests
 {
     public static class WebSocketConnectionExtensions
     {
-        public static async Task<WebSocketConnectionSummary> ExecuteAndCaptureFramesAsync(this IWebSocketConnection connection)
+        public static Task<WebSocketConnectionSummary> ExecuteAndCaptureFramesAsync(this IWebSocketConnection connection)
         {
-            var frames = new List<WebSocketFrame>();
-            var closeResult = await connection.ExecuteAsync(frame => frames.Add(frame.Copy()));
-            return new WebSocketConnectionSummary(frames, closeResult);
+            return Task.Run(async () =>
```

Comment:
Partially it was to help debugging. Without the Task.Run, Parallel Stacks wasn't showing anything really helpful and I was debugging issues between two different Tasks. Once I added the Task.Run, it was much easier to see where one Task was blocked when a particular action occurred (and from what I could tell it wasn't because the timing shifted). I expect I'll remove it before merging. The current head of this PR is somewhat in-progress.